### DISCUSSION TITLE
v0.14b1 test

### DISCUSSION THREAD

**Discussion by harald7** [Louisvdw/dbus-serialbattery#261] [General]
Hi Louis, dank u well for all your work and research. On a raspberry pi 4 / 1.4. I have installed a fresh 2.91 large image. To the raspberry are only 4 jkbms (usal 5 but 1 rs485 module dosen't work anymore) in 16s configuration. 
I did install the latest release (0.13) and then a update to 0.14b1.

All 4 bms pop up in the console.
![Screenshot (89)](https://user-images.githubusercontent.com/115894307/196046552-b4a85064-b82d-4ac9-b9c4-dce5c68c2c3c.png)

The cell voltage of all cells are visible.
![Screenshot (90)](https://user-images.githubusercontent.com/115894307/196046569-7ed3e94e-663f-4ff9-ae85-c8cfa49b7bae.png)

But under Device I can't give an individualistic Name.
![Screenshot (91)](https://user-images.githubusercontent.com/115894307/196047004-d45285fd-0e14-4529-bb70-8650949cd11e.png)

Each bms gets their own vrm instance. 
![Screenshot (93)](https://user-images.githubusercontent.com/115894307/196047642-6222c972-3bdf-4af0-b367-036da6707fcb.png)

The only place where I can give a Name to the bms is under Settings/System setup/ Battery Measurements. But this Name isn't used in the device list of the console or portal
![Screenshot (92)](https://user-images.githubusercontent.com/115894307/196047740-01d76101-244e-4b97-8d8a-c437b6d53d56.png)

Is there any way to get the induvidual names into the device lists?
Should I provide any log?

Thanks a lot in advance!

Cheers
Harald

**Comment by Louisvdw** [comment]
Hi Herald
I'm glad it is all working. Thanks for the report back.
It seems that there is a problem with the custom name in that beta release. Other users have also reported it missing. Hopefully I will have it fixed in the next beta

**Comment by harald7** [comment]
Thanks a lot!

**Reply by merlep** [reply]
Are  all the Cell voltages recorded by the VRM portal?  Are they all visible on the *.csv download from VRM?.

**Reply by Louisvdw** [reply]
The cell voltages are stored in a custom location. Victron does not have a place for them, so they are not in VRM.
Only the min/max cell values are Victron fields.

**Reply by arrow1800** [reply]
@merlep but the highest and lowest cell voltage of **each** pack IS recorded to vrm, just to get all the facts straight:)

**Reply by merlep** [reply]
thanks arrow1800

**Comment by janekx** [comment]
Issue to get 14 beta 1 to work 
Cerbo GX 2.9.1 large
Wget https://github.com/Louisvdw/dbus-serialbattery/releases/download/v0.14beta1/venus-data.tar.gz
tar -zxf venus-data.tar.gz.1 -C /data because downloaded like .1

nano /data/etc/dbus-serialbattery/utils.py 
see Version 14 sub 1 
make changes to my setup 

reboot 

No cell volatges 
no device name to rename 

gone to run sh rc-local to make reinstallocal.sh 

reboot 

still same 
no volatge cels 
no name to rename

**Reply by Louisvdw** [reply]
Log into the GX with SSH and run 
> dbus-spy

Under your serial battery do you see any values for the cells under
*dbus-serialbattery_* /Voltages/Cell1
                                  /Voltages/Cell2
                                  /Voltages/Cell3
                                  etc.

**Reply by janekx** [reply]
No only min and max and diff, 
Cell1 - 16 says -

**Reply by Louisvdw** [reply]
The Cell Voltages page will only show if those values are populated. 
There are 2 reasons why they will not be populated.
1) Your BMS does not support these values
2) Your BATTERY_CELL_DATA_FORMAT = 1 from utils.py have been changed from 1.

So first make sure you have BATTERY_CELL_DATA_FORMAT = 1
Which BMS are you using in your system?

**Reply by janekx** [reply]
the BATTERY_CELL_DATA_FORMAT = 1

Daly 300A Smart BMS FAN RS485 + BT

**Reply by janekx** [reply]
I remeber i have connected the BMS to netebook and it sghows cells in windows DALY app 
maybe diferent bus parsing?

**Reply by janekx** [reply]
[Daly UART_485 Communications Protocol V1 2.pdf](https://github.com/Louisvdw/dbus-serialbattery/files/9900201/Daly.UART_485.Communications.Protocol.V1.2.pdf)

**Comment by merlep** [comment]
I'm a bit confused how to install he 0.14b.  , already have 013 running . Sorry to be particular but this is at the limit of my understanding .

Step 1? - do you delete existing venus-data.tar.gz from previous install?
Step 2  wget https://github.com/Louisvdw/dbus-serialbattery/releases/download/v0.14beta1/venus-data.tar.gz
The  filename downloaded might be a .gz.1 ( according to Janekx) - do you need to rename ?

Step 3 ?? Now this confuses me - do you not have to run the shell  command   sh installrelease.sh as was done with previous install?
Step 4 tar -zxf venus-data.tar.gz -C /data

Thanks

**Reply by Louisvdw** [reply]
These are the 3 steps to follow
https://github.com/Louisvdw/dbus-serialbattery/wiki/How-to-install#install-spesific-versiontesting-option

What happens is if you already have the file venus-data.tar.gz then the wget command will download it again, but as venus-data.tar.gz.1 or after that venus-data.tar.gz.2 which is similar to what Windows will have with it's (1) and (2) file suffix. So it is best to delete the existing venus-data.tar.gz file to make sure it is easier.

installrelease.sh is for installing the latest release, which will be v0.13, so you have to use the spesific version option.

**Reply by Louisvdw** [reply]
PS. You can change wget to force to override the existing command.
 wget -O https://github.com/Louisvdw/dbus-serialbattery/releases/download/v0.14beta1/venus-data.tar.gz

**Reply by havnjero** [reply]
wget -O does not work for me on venus os v.3
otherwise I'm having the same issue with a jkbms

**Comment by merlep** [comment]
I understand that , but whats confusing is the unzipping of the archive
command -
With installed release you use the  sh installrelease.sh command
Now with the beta do you not have to execute a sh ????.sh command at all ??
Just the the tar -zxf venus-data.tar.gz -C /data command
best regards

**Reply by janekx** [reply]
SH is script processing command 
TAR is only extraction
as the driver si installed the tar so replace files for newer version is enought

**Reply by Louisvdw** [reply]
Just different ways to do almost the same thing.
If you go and look at what the sh installrelease.sh do you will see that it also has a wget in the first line and then also do the tar just after that. So this is the same except the installrelease.sh command is packaged to make it easier for people that just want to install the latest release version. If you want to use a spesific version then you need to do those steps manually. 

https://github.com/Louisvdw/dbus-serialbattery/blob/master/etc/dbus-serialbattery/installrelease.sh

**Comment by janekx** [comment]
@Louisvdw do yuo think the data parsing from RS485 is diferent so i cant read cells. I have place the protocol from daly if you can check

**Reply by Louisvdw** [reply]
@janekx the same protocol is used for all UART types for the Daly. So I don't expect that using different ports would change that. What can be an influence is the firmware that the BMS use inside. Daly has many different chips and versions and they are not all the same. My guess would be that this might be where your issue is that you do not see cell values inside. I know there is a way to upgrade Daly firmware, but I don't have much information how and were to get it.

**Comment by merlep** [comment]
I'm sorry but I am not a software engineer ( just a regular engineer ! ) so
my question might seem trivial , !

**Comment by janekx** [comment]
@Louisvdw i have used some serial monitorin software, well not satisfied with it but get this port monitor datas. 
it is monitored when is used PC Daly app and RS485 to usb DALY genuine dongle what is used to connect to Carbo 
[daly.zip](https://github.com/Louisvdw/dbus-serialbattery/files/9937578/daly.zip)

**Reply by lennycb** [reply]
it's the same for me.

Custom Name (Battery -> Device -> Name) --> is not displayed
Cell Voltage GUI display --> is not displayed

In the app I can see the individual cells.
My BMS: Daly 400A Smart BMS RS485 + CAN + BT

**Reply by janekx** [reply]
Seems sames dalys, i have make data logs for @Louisvdw he can analyse and extend posibility
